guides/advanced/deploying.md

```diff
@@ -0,0 +1,87 @@
+# Deploying
+
+Deploying WebRTC applications might be cumbersome.
+Here are a couple of nitpicks you should keep in mind when trying to push your project into production.
+
+### Allow for UDP traffic in your firewall
+
+In most cases, WebRTC uses UDP to exchange audio and video data.
+Therefore, you have to allow UDP traffic in your firewall.
+In Linux-based systems that use `ufw`, you can do this with the following command:
+
+```sh
+ufw allow 50000:60000/udp
+```
+
+Our ICE implementation, by default, uses an ephemeral port range, so it might vary depending on your operating system.
+However, you can specify an exact port range that ICE will use when creating a new peer connection, e.g.:
+
+```elixir
+PeerConnection.start_link(ice_port_range: 50_000..60_000)
+```
+
+### Allow for TCP traffic in your firewall
+
+In some cases, when ICE really cannot find a UDP path, it may fall back to a TCP connection.
+However, since our ICE implementation does not support TCP yet, you don't need to take any extra steps here :)
+
+### Export ports in your docker container
+
+If you are running your application using Docker, we recommend using the `--network host` option.
+If that's not possible (e.g. you are running on macOS), you have to manually export the ports used by ICE, e.g.:
+
+```
+docker run -p 50000-50010/udp myapp
+```
+
+Keep in mind that exporting a lot of ports might take a lot of time or even cause the Docker deamon to timeout.
+That's why we recommend using host's network.
+
+### Choose your cloud provider wisely
+
+Many cloud providers do not offer good support for UDP traffic.
+In such cases, deploying a WebRTC-based application might be impossible.
+We recommend using bare machines that you can configure as you need.
+
+### Enable HTTPS in your frontend
+
+The server hosting your frontend site must have HTTPS enabled.
+This is a requirement for accessing the user's microphone and camera devices.
+
```

Comment:
I'd explicitly say that not using HTTPS (on addresses different than localhost) will result in `navigator.mediaDevices` being `null` because that's really non-obvious.